plugins/koterm.koplugin/main.lua

```diff
@@ -0,0 +1,575 @@
+--[[
+This plugin provides a terminal emulator (VT52 (+some ANSI))
+]]
+
+local Device = require("device")
+
+-- grantpt and friends are necessary (introduced on Android in API 21).
+-- So sorry for the Tolinos with (Android 4.4.x).
+-- Maybe https://f-droid.org/de/packages/jackpal.androidterm/ could be an alternative then.
+if Device:isAndroid() then
+    local A, android = pcall(require, "android")  -- luacheck: ignore
+    local api = android.app.activity.sdkVersion
+    if api < 21 then
+        return
+    end
+end
+
+local Aliases = require("aliases")
+local Dispatcher = require("dispatcher")
+local DataStorage = require("datastorage")
+local Font = require("ui/font")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local MultiConfirmBox = require("ui/widget/multiconfirmbox")
+local ScrollTextWidget = require("ui/widget/scrolltextwidget")
+local SpinWidget = require("ui/widget/spinwidget")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local TermInputText = require("terminputtext")
+local TextWidget = require("ui/widget/textwidget")
+local bit = require("bit")
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local ffi = require("ffi")
+local C = ffi.C
+
+-- for terminal emulator
+ffi.cdef[[
+static const int SIGTERM = 15;
+
+int grantpt(int fd) __attribute__((nothrow, leaf));
+int unlockpt(int fd) __attribute__((nothrow, leaf));
+char *ptsname(int fd) __attribute__((nothrow, leaf));
+pid_t setsid(void) __attribute__((nothrow, leaf));
+
+static const int TCIFLUSH = 0;
+int tcdrain(int fd) __attribute__((nothrow, leaf));
+int tcflush(int fd, int queue_selector) __attribute__((nothrow, leaf));
+]]
+
+local CHUNK_SIZE = 80 * 40 -- max. nb of read bytes (reduce this, if taps are not detected)
+
+local Terminal = WidgetContainer:new{
+    name = "KOTerm",
```

Comment:
`name = "koterm"` (not sure, but I think I remember it should be the same as in _meta.lua)